### DISCUSSION TITLE
Multiple actions in one turn

### DISCUSSION THREAD

**Discussion by maxkonrad** [bcollazo/catanatron#286] [Q&A] [answered]
In current code, can agents do multiple actions in a turn? What is the general view on this? What do you guys think?

**Comment by bcollazo** [comment]
Hey @maxkonrad , yes bots/players can do multiple actions in one turn. How this is implemented is that _END_TURN_ is one of actions. And so carrying out an action (like BUILD_SETTLEMENT, or PLAY_YEAR_OF_PLENTY) doesn't advance the turn of the current player. Not until END_TURN is called by the player. Is this what you mean? LMK!

**Comment by maxkonrad** [comment]
Oh ok. Thanks so much. I was thinking END_TURN is always -inf. Get possible actions also gets END_TURN right?

**Comment by maxkonrad** [comment]
Sorry for my dumbness. Can I use negamax in multi-player games instead of alphabeta prunned minimax?

**Reply by bcollazo** [reply, marked as answer]
Hey, first time I hear about negamax. From quick browsing in Wikipedia it seems its only for 2-player games? But it'd be good to explore it nonetheless for the 1v1 Catan variant. Give it a try! ChatGPT might be able to generate an initial version given the AlphaBeta implementation we already have. Good luck!

**Reply by maxkonrad** [reply]
Thanks so much. Really appreciated!
